app/javascript/controllers/audio_player_controller.js

```diff
@@ -0,0 +1,104 @@
+import { Controller } from "@hotwired/stimulus"
+import convertSecondsToDuration from "util/convertSecondsToDuration"
+
+export default class extends Controller {
+  static targets = ["audio", "duration", "progress", "progressBar"]
+
+  connect() {
+    this.playing = false
+    this.seeking = false
+    this.originalDuration = this.durationTarget.innerHTML
+    this.bindMouseMove = this.mouseMove.bind(this)
+    this.bindMouseUp = this.mouseUp.bind(this)
```

Comment:
Which `this` do each `this` refer to?

Reply:
`this` is always the instance of the controller we're in.  I'm creating a new version of that method, where within it, `this` still refers to the controller.  Same as right here.

If I didn't do that, and called:

```js
window.addEventListener("mousemove", this.mouseMove)
```

Then inside mouse move:

```js
mouseMove(event) {
  // an error would happen on the next line, because "this" is no longer a stimulus controller
  this.seekProgressBar(event.x)
}
```